### ISSUE TITLE
Authentication failure using <user>/<password or personal access token> in Bitbucket server 

### ISSUE THREAD

**Issue by Enrice** [jenkinsci/bitbucket-branch-source-plugin#904]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```
jenkins 2.479.1
plugin version [895.v15dc41668f03](https://plugins.jenkins.io/cloudbees-bitbucket-branch-source)
bitbucket data center 8.19.9
  ```

</details>

This is a Multibranch Pipeline Events log. As can be seen, authentication works for listing the branches. But the git fetch does not.

```
[Tue Nov 05 12:34:20 CET 2024] Received com.cloudbees.jenkins.plugins.bitbucket.hooks.NativeServerPushHookProcessor$HeadEvent UPDATED event from *** ⇒ https://***/bitbucket-scmsource-hook/notify with timestamp Tue Nov 05 12:34:15 CET 2024
Connecting to https://*** using ***/****** (jenkins' Bitbucket technical user)
Repository type: Git
Looking up ***/*** for branches
Checking branch release/np-1.31 from ***/***
      ‘Jenkinsfile’ found
    Met criteria
Changes detected: release/np-1.31 (7ed4c9ea362282c445dc1d793d1cf633967a08bf → 17b07cbe0f997b8654b8f1d440e950046d1f53d8)
ERROR: Exception: java.io.IOException: hudson.plugins.git.GitException: Command "git fetch --tags --force --progress --prune -- origin +refs/heads/release/np-1.31:refs/remotes/origin/release/np-1.31" returned status code 128:
stdout: 
stderr: fatal: Authentication failed for '***'


Scheduled build for branch: release/np-1.31

  1 branches were processed (query completed)
```

### What Operating System are you using (both controller, and any agents involved in the problem)?

Rocky Linux 9

### Reproduction steps

see log

### Expected Results

git fetch should be authenticated

### Actual Results

it is not

**Comment by nfalco79** [comment]
Please report the actual credentials type are you using (ssh, oauth2, app password)?

**Comment by nfalco79** [comment]
Do you have cache enanled?

**Comment by Enrice** [comment]
credentials type is user/password, while password is actually an http access token with admin rights on the project.
last plugin version that worked was 886.v44cf5e4ecec5

no caching of the branch list
when triggered manually the same happens.

**Comment by Enrice** [comment]
I tried with [906.vedf430cb_4481](https://updates.jenkins.io/download/plugins/cloudbees-bitbucket-branch-source/906.vedf430cb_4481/cloudbees-bitbucket-branch-source.hpi) now, same.

**Comment by nfalco79** [comment]
> credentials type is user/password, while password is actually an http access token with admin rights on the project.

Do you mean that you have configured an OAuth2 credential with secretId and secretPassword as a Username/Password Jenkins credentials or in the Jenkins credential you have setup as user x-auth-token/<token value> ?

The first one works and is documentation describe, the second one no.
We have configured OAuth2 for bibtucket cloud and scan/git clone works so I something is missing to reproduce this issue.

**Comment by nfalco79** [comment]
Changes in release https://github.com/jenkinsci/bitbucket-branch-source-plugin/releases/tag/895.v15dc41668f03 are:
1) one re-throw exception that was eat
2) other one is about the key in cache (but in your case is disabled)
3) Update jenkins dependencies (that you will get also if this plugin did not move forward)

**Comment by Enrice** [comment]
> > credentials type is user/password, while password is actually an http access token with admin rights on the project.
> 
> Do you mean that you have configured an OAuth2 credential with secretId and secretPassword as a Username/Password Jenkins credentials or in the Jenkins credential you have setup as user x-auth-token and as password <token value> ?
> 
> The first one works and follow the documentation, the second one no.
> 
> We have configured OAuth2 for bibtucket cloud and scan/git clone works so I something is missing to reproduce this issue.

configured token for CI user in bitbucket here:
![Image](https://github.com/user-attachments/assets/6a594f6f-c086-419c-96b6-85795849025b)

credentials defined in jenkins:
![Image](https://github.com/user-attachments/assets/2e6d22b1-fff6-4d1d-a138-c00845dc3b9f)

used in the job:
![Image](https://github.com/user-attachments/assets/21d45e3a-bce5-4db2-9666-fabcd7ef96a6)


Bitbucket Datacenter here. No cloud.
Plz notice I updated latest working version above which was 886, all later ones fail with mentioned error.

**Comment by nfalco79** [comment]
Clonse since this is not a bug but a requested feature.
See issue #716

**Comment by Enrice** [comment]
please reopen! I think you mixed sth. up. this IS a bug, not a feature request.

reason:
I am NOT!!! using a repository or project token, which would have to be used with bearer authentication, but a PERSONAL ACCESS TOKEN, which is used instead of the password only. This is totally transparent to jenkins or any plugin and will be usable in all basic authentication scenarios.

see again:
![Image](https://github.com/user-attachments/assets/b3e06ed6-40f6-4617-9f5c-92762e490cff)

**Comment by nfalco79** [comment]
The AppPassword for bitcucket cloud works and it's a username/password couple credentials in place of real user/password. In Bitbucket cloud REST API does not allow the use of real user/password.
Unfortunately the guide in the README.MD to startup a local server 
`server: atlas-run-standalone -u 6.3.0 --product bitbucket --version 5.2.0 --data-version 5.2.0`
does not work anymore (I used at the time of my first contribution to this plugin). Altassian SDK is not able to run server > 7.21 on windows so I'm tring using docker image to run a local bitbucket server 8.16

**Comment by Enrice** [comment]
plz notice the update in my Environment above. I use bitbucket data center 8.19.9 (LTS)

**Comment by Enrice** [comment]
just to narrow down the possibly causing changes: first broken version was [887.va_d359b_3d2d8d](https://updates.jenkins.io/download/plugins/cloudbees-bitbucket-branch-source/887.va_d359b_3d2d8d/cloudbees-bitbucket-branch-source.hpi)

**Comment by nfalco79** [comment]
@Enrice could you please test the incremental build produced by PR #924 ?
Should fix this issue, at least use the given credentials as is for username /password credential
.
https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/plugins/cloudbees-bitbucket-branch-source/920.v068d50e9e2f2/

**Comment by Enrice** [comment]
Thank you. Actually fixes both - so #863 was just a follow-up error!

**Comment by nfalco79** [comment]
Thanks very much.
Let me the time to implements some unit test and I will release the fix.
